experiments/targeting.go

```diff
@@ -0,0 +1,298 @@
+package experiments
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"strings"
+)
+
+type AnyNode struct {
+	children []Targeting
+}
+
+func NewAnyNode(input interface{}) (Targeting, error) {
+	inputNodes, ok := input.([]interface{})
+	if !ok {
+		return nil, TargetingNodeError("input to AnyNode expects an array")
+	}
+	children := make([]Targeting, len(inputNodes))
+	for i, node := range inputNodes {
+		node, err := parseNode(node)
+		if err != nil {
+			return nil, err
+		}
+		children[i] = node
+	}
+	return &AnyNode{
+		children: children,
+	}, nil
+}
+
+func (n *AnyNode) Evaluate(inputs map[string]interface{}) bool {
+	for _, node := range n.children {
+		if node.Evaluate(inputs) {
+			return true
+		}
+	}
+	return false
+}
+
+type AllNode struct {
+	children []Targeting
+}
+
+func NewAllNode(input interface{}) (Targeting, error) {
+	inputNodes, ok := input.([]interface{})
+	if !ok {
+		return nil, TargetingNodeError("input to AnyNode expects an array")
+	}
+	children := make([]Targeting, len(inputNodes))
+	for i, node := range inputNodes {
+		node, err := parseNode(node)
+		if err != nil {
+			return nil, err
+		}
+		children[i] = node
+	}
+	return &AllNode{
+		children: children,
+	}, nil
+}
+
+func (n *AllNode) Evaluate(inputs map[string]interface{}) bool {
+	for _, node := range n.children {
+		if !node.Evaluate(inputs) {
+			return false
+		}
+	}
+	return true
+}
+
+type EqualNode struct {
+	field  string
+	values []interface{}
+}
+
+func NewEqualNode(inputNodes map[string]interface{}) (Targeting, error) {
+	if len(inputNodes) != 2 {
+		return nil, TargetingNodeError("EqualNode expects exactly two fields")
+	}
+	acceptedKey, ok := inputNodes["field"]
+	if !ok {
+		return nil, TargetingNodeError("EqualNode expects input key 'field'")
+	}
+
+	value, valueOK := inputNodes["value"]
+	values, valuesOK := inputNodes["values"]
+	if !valueOK && !valuesOK {
+		return nil, TargetingNodeError("EqualNode expects input key 'value' or 'values'.")
+	}
+
+	var acceptedValues []interface{}
+	if valueOK {
+		acceptedValues = []interface{}{value}
+	}
+	if valuesOK {
```

Comment:
it seems like the intention here is to favor `values` over `value`?  If that is the case, I think it's clearer to write that as:

```go
if valuesOK {
	acceptedValues = values.([]interface{})
} else if valueOK {
	acceptedValues = []interface{}{value}
}
```

Reply:
Yes, that's what I got from reading the baseplate.py code, that looks good.